internal/operator-controller/rukpak/convert/helm.go

```diff
@@ -18,11 +19,27 @@ type BundleToHelmChartConverter struct {
 }
 
 func (r *BundleToHelmChartConverter) ToHelmChart(bundle source.BundleSource, installNamespace string, watchNamespace string) (*chart.Chart, error) {
+	config := make(map[string]interface{})
+
+	if watchNamespace != "" {
+		config["watchNamespace"] = watchNamespace
+	}
+
 	rv1, err := bundle.GetBundle()
 	if err != nil {
 		return nil, err
 	}
 
+	if _, present := config["watchNamespace"]; !present {
+		config["watchNamespace"] = r.deriveDefaultWatchNamespace(&rv1, installNamespace)
+	}
+
+	// Derive target namespaces from watchNamespace
+	targetNamespaces, err := r.deriveTargetNamespaces(config)
+	if err != nil {
+		return nil, fmt.Errorf("error deriving target namespaces: %w", err)
+	}
+
```

Comment:
Instead of this, create an options slice and add the certificate provider and only add the TargetNamespace option if the configuration is defined.